src/FileWriter.php

```diff
@@ -25,21 +25,30 @@ final class FileWriter
      */
     public static function writeFile($file, $content, $chmod = 0666)
     {
-        $tmp = tempnam(sys_get_temp_dir(), md5($file));
+        $dir = dirname($file);
+        $tmp = tempnam($dir, 'wsw');
+
+        if ($tmp === false) {
+            throw Exception\PermissionException::unableToCreateTemporaryFile($tmp);
```

Comment:
It's not necessarily a permission issue. It can be a full disk or another hardware failure. A `RuntimeException` or something in this area might be more conservative.